docs/design-patterns.mdx

```diff
@@ -74,75 +180,279 @@ and in the transaction that is used to deploy it:
+        let temporaryVault <- vaultRef.withdraw(amount: 10.0)
+        let self.buyerCollectionRef = acct.borrow(from: /storage/Collection)
+
+        // purchase, supplying the buyers collection reference
+        saleRef.purchase(tokenID: 1, recipient: self.buyerCollectionRef, buyTokens: <-temporaryVault)
+
+    }
+    post {
+        // verify that the buyer now owns the NFT
+        self.buyerCollectionRef.idExists(1) == true: "Bought NFT ID was not deposited into the buyers collection"
     }
+}
+```
 
-    pub resource BResource {
-        pub var c: UInt64
-        pub var d: string
+## Avoid Excessive Load and Save Storage Operations (Prefer in-place Mutations)
 
-        pub fun generateReport() BReportStruct {
-            return BReportStruct(c: c, d: d)
-        }
+### Problem
+
+When modifying data in account storage, `load()` and `save()` are costly operations.
+This can quickly cause your transaction to reach the gas limit or slow down the network.
+
+### Solution
+
+For making modifications to values in storage or accessing stored objects,
+`borrow()` should always be used to access them instead of `load` or `save` unless absolutely necessary.
+`borrow()` returns a reference to the object at the storage path instead of having to load the entire object.
+This reference can be assigned to or can be used to access fields or call methods on stored objects.
+
+### Example
+
+```cadence
+// Loads and stores a resource to use it
+// BAD
+transaction {
+
+    prepare(acct: AuthAccount) {
+
+        // Removes the vault from storage, a costly operation
+        let vault <- acct.load<@ExampleToken.Vault>(from: /storage/exampleToken)
+
+        // Withdraws tokens
+        let burnVault <- vault.withdraw(amount: 10)
+
+        destroy burnVault
+
+        // Saves the used vault back to storage, another costly operation
+        acct.save(to: /storage/exampleToken)
 
-        init(c: UInt64, d: string) {
-            self.c = c
-            self.d = d
-        }
     }
 }
-...
-import A from 0xA
 
-pub fun main(): A.BReport {
-    let b: AContract.BResource // Borrow the resource
-    let report = b.generateReport()
-    return b
+// Uses borrow instead to avoid costly operations
+// GOOD
+transaction {
+
+    prepare(acct: AuthAccount) {
+
+        // Borrows a reference to the stored vault, much less costly operation
+        let vault <- acct.borrow<&ExampleToken.Vault>(from: /storage/exampleToken)
+
+        let burnVault <- vault.withdraw(amount: 10)
+
+        destroy burnVault
+
+        // No `save` required because we only used a reference
+    }
 }
 ```
+
+# Capabilities
+
+## Capability Receiver
+
+### Problem
+
+An account must be given a [capability](/cadence/language/capability-based-access-control/)
+to a resource or contract in another account but a single transaction with both accounts authorizing it
+cannot be easily produced. This prevents a single transaction from fetching the capability
```

Comment:
It might not be clear to the reader why the transaction must be signed by both parties. Maybe explain:

```suggestion
to a resource or contract in another account. To create, i.e. link the capability, the transaction must be signed by a key which has access to the target account.
To transfer / deliver the capability to the other account, the transaction also needs write access to that one.
It is not as easy to produce a single transaction which is authorized by two accounts as it is to produce a typical transaction which is authorized by one account. 
This prevents a single transaction from fetching the capability
```